sigh/lib/sigh/resign.rb

```diff
@@ -72,7 +72,7 @@ def resign(ipa, signing_identity, provisioning_profiles, entitlements, version,
       puts(`#{command}`)
       command_return_value = $?.to_i
 
-      `security list-keychains -s #{current_keychains}` if current_keychains
+      `security list-keychains -s #{previous_keychains}` if previous_keychains
```

Comment:
Just noticed that this will _always_ be true if `keychain_path` exists, true?  
A bit less stuff executed if the if block in lines 25+ was rewritten to only add the new keychain (and backup the previous ones) if it is not already in there. Does this make sense or did I misudnerstand your code?

Reply:
Yes you are correct.
So the if block could be written like this. This would really only execute each step if it is truly needed, right?
```
if keychain_path
    keychain_path = File.expand_path(keychain_path)

    current_keychains = `security list-keychains`
    current_keychains.delete!("\n")

    unless current_keychains.include?(keychain_path)
        previous_keychains = current_keychains 
        `security list-keychains -s #{current_keychains} '#{keychain_path}'`
    end
end
```

Reply:
That looks like it would work to me. (But I am much better at pointing out stuff that might not work than to actually write working ruby code myself, so... 🍶 )

Reply:
Done